src/mlpack/methods/ann/layer/lookup_impl.hpp

```diff
@@ -21,20 +21,29 @@ namespace ann /** Artificial Neural Network. */ {
 
 template <typename InputDataType, typename OutputDataType>
 Lookup<InputDataType, OutputDataType>::Lookup(
-    const size_t inSize,
-    const size_t outSize) :
-    inSize(inSize),
-    outSize(outSize)
+    const size_t vocabSize,
+    const size_t embeddingSize) :
+    vocabSize(vocabSize),
+    embeddingSize(embeddingSize)
 {
-  weights.set_size(outSize, inSize);
+  weights.set_size(embeddingSize, vocabSize);
 }
 
 template<typename InputDataType, typename OutputDataType>
 template<typename eT>
 void Lookup<InputDataType, OutputDataType>::Forward(
     const arma::Mat<eT>& input, arma::Mat<eT>& output)
 {
-  output = weights.cols(arma::conv_to<arma::uvec>::from(input) - 1);
+  const size_t seqLength = input.n_rows;
+  const size_t batchSize = input.n_cols;
+
+  output.set_size(embeddingSize * seqLength, batchSize);
+
+  for (size_t i = 0; i < batchSize; ++i)
+  {
+    output.col(i) = arma::vectorise(weights.cols(
+        arma::conv_to<arma::uvec>::from(input.col(i)) - 1));
+  }
```

Comment:
Do we have an example of how `Lookup<>` is supposed to be used?  It seems to me like it would expect an input point to be one-dimensional, and it would just look up the corresponding column in `weights`.  When the batch size is greater than one with a feed-forward network, then `input` will have number of columns equal to the batch size and number of rows equal to the dimensionality.

So, if this expects a dimensionality of 1, then the `arma::conv_to<arma::uvec>::from(input)` will create an `arma::uvec` of indices, one for each point in the batch.  Then `weights.cols()` will simply produce the corresponding column of `weights` for each individual input index.

If I've understood that correctly, then it seems like this already would support a batch size greater than 1.  Please, correct me if I've misunderstood something though. :+1:

Reply:
If `input` is matrix type with shape `(sequenceLength, batchSize)` with `batchSize > 1`, the statement `arma::conv_to<arma::uvec>::from(input)` will not be able to convert it to `uvec` type and throw error. 

Now if I use `arma::vectorise(arma::conv_to<arma::umat>::from(input))`, the resultant vector is of shape `(sequenceLength * batchSize, 1)`. And then whichever numbers are there in this vector, those columns will be selected from the weight. So the shape of the `output` will be `(weights.n_rows, sequenceLength * batchSize)`. This is distorting the placement of batches for the next layers.

Instead if the `output` has shape `(weights.n_rows * sequenceLength, batchSize)`, the number of columns i.e. `batchSize` is preserved for next layers.

This is what I have tried to do in this PR.

Reply:
I agree with Mrityunjay, the input can be multidimensional. For instance each dataset record could represent a sequence of words. In this case `arma::conv_to<arma::uvec>::from(input)` would raise an exception.